core/challenges.go

```diff
@@ -5,31 +5,73 @@
 
 package core
 
+import (
+	"encoding/json"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/letsencrypt/go-jose"
+)
+
+func newChallenge(challengeType string, accountKey *jose.JsonWebKey) (Challenge, error) {
+	ak := AuthorizedKey{
+		Token: NewToken(),
+		Key:   accountKey,
+	}
+
+	jsonAK, err := json.Marshal(ak)
```

Comment:
We marshal this ak three times in pa.ChallengesFor.

Why can't we wait to serialize the ak when we serialize the Challenge?